src/rules-engine/triggers/TimeTrigger.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.*
+ */
+
+const Events = require('../Events');
+const Trigger = require('./Trigger');
+
+/**
+ * An abstract class for triggers whose input is a single property
+ */
+class TimeTrigger extends Trigger {
+  constructor(desc) {
+    super();
+    this.time = desc.time;
+    this.sendStateChanged = this.sendStateChanged.bind(this);
+  }
+
+  /**
+   * @return {TriggerDescription}
+   */
+  toDescription() {
+    return Object.assign(
+      super.toDescription(),
+      {time: this.time}
+    );
+  }
+
+  async start() {
+    this.scheduleNext();
+  }
+
+  scheduleNext() {
+    // TODO(hobinjk): support sunrise and sunset
+    let parts = this.time.split(':');
+    let hours = parseInt(parts[0], 10);
+    let minutes = parseInt(parts[1], 10);
+
+    let nextTime = new Date();
+    nextTime.setHours(hours);
+    nextTime.setMinutes(minutes);
+    nextTime.setSeconds(0);
+    nextTime.setMilliseconds(0);
```

Comment:
This is probably fine for this release, but we should figure out what to do if the wall time changes (like at daylight savings time changes and/or the user just changes the time).

I only looked at TimeTrigger.js since the other files appear to be mostly UI stuff.

Reply:
Good point! As far as I can tell from playing around in the console with Date this will always trigger at a time when the wall clock is correct unless the time doesn't exist. In the spring forward case where there can be a nonexistent time it will silently advance an hour. In the fall back case it looks like it goes for the first instance of the time.